src/Operation/Factory/SubresourceOperationFactory.php

```diff
@@ -137,9 +131,9 @@ private function computeSubresourceOperations(string $resourceClass, array &$tre
                 $operation['path'] .= sprintf('/%s%s', $this->pathSegmentNameGenerator->getSegmentName($property, $operation['collection']), self::FORMAT_SUFFIX);
             }
 
-            $tree[$visiting] = $operation;
+            $tree[$operation['route_name']] = $operation;
 
-            $this->computeSubresourceOperations($subresourceClass, $tree, $rootResourceClass, $operation, $visiting);
+            $this->computeSubresourceOperations($subresourceClass, $tree, $rootResourceClass, $operation, array_merge($visited, [$visiting]));
```

Comment:
Oh sorry, I missed the loop! BTW it's better like this with an associative array! 👍